File: Baekjoon/Python/창고 다각형.py
import sys
inpyut = sys.stdin.readline

n = int(input().rstrip())
stick = [0]*1001
max_h = 0
max_idx = 0
for i in range(n):
    l, h = map(int, input().split())
    stick[l] = h
    if max_h < h:
        max_idx = l
        max_h = h

cur_L = 0
ans = 0
for i in range(max_idx+1):
    cur_L = max(cur_L,stick[i])
    ans += cur_L

cur_R = 0
for i in range(1000,max_idx,-1):
    cur_R = max(cur_R,stick[i])
    ans += cur_R

print(ans)

File: Baekjoon/Java/창고 다각형.java
import java.io.*;
import java.util.*;


public class Main {
    static int max_h;
    static int max_idx;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] stick = new int[1001];
        for (int i : stick) {
            stick[i] = 0;
        }
        int n = Integer.parseInt(br.readLine());
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int l = Integer.parseInt(st.nextToken());
            int h = Integer.parseInt(st.nextToken());
            stick[l] = h;
            if (max_h < h) {
                max_h = h;
                max_idx = l;
            }
        }

        int cur_L = 0;
        int cur_R = 0;
        int ans = 0;
        for (int i = 0; i <= max_idx; i++) {
            cur_L = Math.max(cur_L, stick[i]);
            ans += cur_L;
        }
        for (int i = 1000; i > max_idx; i--) {
            cur_R = Math.max(cur_R, stick[i]);
            ans += cur_R;
        }

        System.out.println(ans);


    }
}